core/drivers/crypto/se050/core/cipher.c

```diff
@@ -16,8 +16,10 @@
 static TEE_Result do_init(struct drvcrypt_cipher_init *dinit)
 {
 	struct crypto_cipher_ctx *ctx = dinit->ctx;
+	TEE_OperationMode mode = dinit->encrypt ?
```

Comment:
Well spotted!
I'd prefer writing this out with a proper if-statement instead of using the `?` operator.

Reply:
I am wondering if this is an issue on itself or just the API semantics; perhaps dinit->encrypt should be renamed to something like mode instead? @cneveux ?


Also doesnt it have the same effect on the caam_cipher.c? 

`        CIPHER_TRACE("Action %s", dinit->encrypt ? "Encrypt" : "Decrypt");                                                                  
`

Reply:
Hi Jorge. I see this as just an issue confined to the se050 cipher driver -- i.e. incorrect handling of the inbound `encrypt` field when passing it through to the `crypto_cipher_ops` `init `function in the next layer down.

The reason it doesn't affect the caam driver is because the caam driver's `caam_cipher_initialize `implementation uses `cipherdata `as its context (instead of `crypto_cipher_ops` in the se050) and doesn't offload the processing to a lower layer. It therefore works with the encrypt field consistently as a bool instead of trying to transform it to TEE_OperationMode.

There is indeed a larger question of whether `drvcrypt_cipher_init `should have used a `TEE_OperationMode `instead of a bool for the operation field, but I'm not sure I can answer that.

Reply:
Hi Jorge. I see this as just an issue confined to the se050 cipher driver -- i.e. incorrect handling of the inbound encrypt field when passing it through to the `crypto_cipher_ops` `init` function in the next layer down.

The reason it doesn't affect the caam driver is because the caam driver's `caam_cipher_initialize` implementation uses `cipherdata` as its context (instead of `crypto_cipher_ops` in the se050) and doesn't offload the processing to a lower layer. It therefore works with the encrypt field consistently as a `bool` instead of trying to transform it to `TEE_OperationMode`.

There is indeed a larger question of whether `drvcrypt_cipher_init` should have used a `TEE_OperationMode` instead of a bool for the operation field, but I'm not sure I can answer that

Reply:
Hi Dave, yes I understand the point. I still think the API is misleading (I remember thinking the same when I did this bit of code). anyhow, since you got Jen's ack not much more from me to say.
thanks for the change.

Reply:
Feel free to propose enhancements to the API in another PR.

Reply:
@jenswi-linaro ok will do - we still need to add more changes (some of them already in one of my trees) plus support for the imx7ulp i2c ; it was just that since this PR is not addressing a current functional issue (just the interface certainly fixing a future problem) perhaps we could have done it here. but sure no problem, I'll do that.

Reply:
Not sure if my ack is required but just in case

`Acked-by: Jorge Ramirez-Ortiz <jorge@foundries.org>`

Reply:
@ldts sure, review/ack tags are always helpful and appreciated.